sovrin_client/agent/jsonpickle_util.py

```diff
@@ -0,0 +1,64 @@
+import jsonpickle
+
+from anoncreds.protocol.types import PublicKey, RevocationPublicKey, \
+    SecretKey, RevocationSecretKey, AccumulatorSecretKey
+from anoncreds.protocol.utils import toDictWithStrValues, fromDictWithStrValues
+
+DATA_KEY = 'py/integer-element'
+
+
+class CommonIntegerElementHandler(jsonpickle.handlers.BaseHandler):
+    def flatten(self, obj, data):
+        data[DATA_KEY] = toDictWithStrValues(obj)
+        return data
+
+    def restore(self, obj):
+        dict = fromDictWithStrValues(obj[DATA_KEY])
+        return self._restore(dict)
+
+    def _restore(self, dict):
+        raise NotImplemented
+
+
+class PublicKeyHandler(CommonIntegerElementHandler):
```

Comment:
Maybe we can use `toStrDict` and `fromStrDict` methods (https://github.com/evernym/anoncreds/blob/master/anoncreds/protocol/types.py) instead?
How does "N", "Rms" and other values in dict get there? Is it like str() of PublicKey's fields, or toStrDict was called before?
If toStrDict wad called to obtain a dict from an anoncreds object, then it must be de-serialized back to anoncreds object by calling fromStrDict